lib/buffer.js

```diff
@@ -663,13 +662,24 @@ Buffer.prototype[customInspectSymbol] = function inspect(recurseTimes, ctx) {
     str += ` ... ${remaining} more byte${remaining > 1 ? 's' : ''}`;
   // Inspect special properties as well, if possible.
   if (ctx) {
+    let extras = false;
     const filter = ctx.showHidden ? ALL_PROPERTIES : ONLY_ENUMERABLE;
-    str += getOwnNonIndexProperties(this, filter).reduce((str, key) => {
-      // Using `formatProperty()` expects an indentationLvl to be set.
-      ctx.indentationLvl = 0;
-      str += `, ${formatProperty(ctx, this, recurseTimes, key, kObjectType)}`;
-      return str;
-    }, '');
+    const obj = getOwnNonIndexProperties(this, filter).reduce((obj, key) => {
+      extras = true;
+      obj[key] = this[key];
+      return obj;
+    }, Object.create(null));
+    if (extras) {
+      if (this.length !== 0)
+        str += ', ';
+      // '[Object: null prototype] {'.length === 26
+      // This is guarded with a test.
+      str += utilInspect(obj, {
+        ...ctx,
+        breakLength: Infinity,
+        compact: true
+      }).slice(27, -2);
```

Comment:
Is there a bit of vulnerable if we hard coded `27` and `-2`  here ?

Reply:
Yes, it is brittle but I do not see a good alternative to it currently. The way we rely on the output is by using `compact: true` and `breakLength: Infinity`. That should always result in exactly this output and we have tests to guard against future failures. That's why I have the comment above the inspect call.